frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -675,465 +919,376 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = ({
-        const groupNumIndex = pathInfoList.indexOf(groupInfo);
-        const groupIndex = group.split(groupName.concat(':'))[1];
-        const newGroupName = groupName
-          .concat('[')
-          .concat(groupIndex)
-          .concat(']');
-
-        // Generate new capabilities
-        const arrayFieldGroupCapability = newField.capabilities.find((c) =>
-          c.startsWith(SpecCapability.arrayFieldGroup),
-        );
-        const newArrayFieldGroupCapability = !_.isUndefined(
-          arrayFieldGroupCapability.split(groupName.concat(':'))[1],
-        )
-          ? _.dropRight(
-              newField.capabilities
-                .find((c) => c.startsWith(SpecCapability.arrayFieldGroup))
-                .split(':'),
-            )
-              .join(':')
-              .concat(':')
-              .concat(groupIndex)
-          : arrayFieldGroupCapability.concat(':').concat(groupIndex);
-
-        newField.capabilities.splice(
-          _.findIndex(newField.capabilities, (c: string) =>
-            c.startsWith(SpecCapability.arrayFieldGroup),
-          ),
-          1,
-          newArrayFieldGroupCapability,
-        );
-        // Generate new group name part in path
-        pathInfoList.splice(groupNumIndex, 1, newGroupName);
-        newField.path = pathInfoList.join('.');
-
-        // Add new fieldValue
-        newFormValues[newField.path] = defaultValueFor(newField);
-
-        // Add new field
-        newFields.push(newField);
-      });
-      setFormValues(newFormValues);
-      setFields(newFields);
-    }
+    setFields((currentFields) => [...currentFields, ...addedFields]);
+    setFormValues((currentFormValues) => ({ ...currentFormValues, ...addedFormValues }));
   };
 
-  const removeFieldsFromGroup = (fieldList, groupName) => {
-    let newFormValues = _.cloneDeep(formValues);
-    let newFieldList = _.cloneDeep(fields);
-    let fieldsInGroup = [];
-
-    if (_.isArray(fieldList) && !_.isEmpty(fieldList) && !_.isNil(groupName)) {
-      fieldsInGroup = newFieldList.filter((f) =>
-        f.capabilities.some((c) => {
-          if (c.startsWith(SpecCapability.arrayFieldGroup)) {
-            return c.split(SpecCapability.arrayFieldGroup)[1].includes(groupName);
-          }
-          return false;
-        }),
-      );
-
-      _.forEach(fieldList, () => {
-        // Remove related formValue
-        const key = _.findLastKey(newFormValues, (v, k) =>
-          k.split('.').some((p) => p.includes(groupName)),
-        );
-        newFormValues = _.omit(newFormValues, [key]);
-
-        // Remove related fields
-        newFieldList = _.without(newFieldList, fieldsInGroup.pop());
-      });
-
-      setFormValues(newFormValues);
-      setFields(newFieldList);
-    }
-  };
+  // Removes fields corresponding to an  arrayFieldGroup index from the form
+  const removeArrayFieldGroup = (fieldLists: OperandField[][], indexToRemove: number) => {
```

Comment:
Same as `addArrayFieldGroup`, just unifying the previous helpers into a single function and updating the logic to make syncing data easier.